google/auth/external_account.py

```diff
@@ -85,6 +86,7 @@ def __init__(
         default_scopes=None,
         workforce_pool_user_project=None,
         universe_domain=_DEFAULT_UNIVERSE_DOMAIN,
+        metrics_options=None,
```

Comment:
Maybe not expose `metric_options` since it's internal details?

Reply:
Needs to be exposed either in the constructor or via an setter, see reply on comment below.

Reply:
I think a private setter might work better since metrics are internal details.

Reply:
Ok, just to make sure I understand, you're suggesting a private setter that is used here like its public (the original external account credential object would be calling it on the source credential after the source credential is created to pass to the impersonated object) but just marked private so users know they shouldn't be crafting/setting their own metrics options

Reply:
yes, that's right. You can introduce a `def _with_metric_options(options)` method here to return a copy of the object but with the given options as `self._metric_options` (i.e. first create a copy of the object, then set `self._metric_options = options`). In the constructor, you can then simply set `self._metric_options = self._create_default_metrics_options()`.